src/ValueResolver/NodeExpressionResolver.php

```diff
@@ -278,6 +279,11 @@ protected function resolveExprClassConstFetch(Expr\ClassConstFetch $node)
         $refClass     = $this->fetchReflectionClass($classToReflect);
         $constantName = ($node->name instanceof Expr\Error) ? '' : $node->name->toString();
 
+        if ($node->class instanceof Name && $node->class->isSpecialClassName()) {
+            $this->isConstant = true;
+            return $node->class . '::' . $constantName;
+        }
```

Comment:
@lisachenko special class name: `self::*, parent::*, static::*` is now printed as is.

Reply:
I will look more why the behaviour is different between in ReflectionMethod vs ReflectionClass.

Reply:
Looking directly on `Expr\ClassConstFetch` seems got invalid result, as it doesn't know what it come from (its parent node), I guess we can replace it with `PhpParser` Printer class instead.

Reply:
using `ParentConnectingVisitor` seems can be solution, but need to verify `nodeLevel` https://github.com/goaop/parser-reflection/pull/127/commits/60194311af8fd1859a1c92a991be45290b055e9c

I will look more for append, eg, via concat with expanded test fixture

Reply:
Note for different result start from php 8.1 https://3v4l.org/IoXSq

![Screenshot 2024-01-18 at 23 19 14](https://github.com/goaop/parser-reflection/assets/459648/46576448-bd71-4996-9ac5-1059755883e3)